packages/mobile/src/fiatExchanges/FiatExchangeAmount.tsx

```diff
@@ -0,0 +1,310 @@
+  currencyCode: CURRENCIES[CURRENCY_ENUM.DOLLAR].code,
+}
+
+const useDollarAmount = (currency: CURRENCY_ENUM, amount: BigNumber) => {
+  const localExchangeRate = useSelector(getLocalCurrencyExchangeRate)
+  const localCurrencyCode = useLocalCurrencyCode()
+  const exchangeRatePair = useSelector(exchangeRatePairSelector)
+
+  if (currency === CURRENCY_ENUM.DOLLAR) {
+    return convertDollarsToMaxSupportedPrecision(
+      (!amount.isNaN() &&
+        convertLocalAmountToDollars(amount, localCurrencyCode ? localExchangeRate : 1)) ||
+        new BigNumber('0')
+    )
+  } else {
+    const exchangeRate = getRateForMakerToken(
+      exchangeRatePair,
+      CURRENCY_ENUM.DOLLAR,
+      CURRENCY_ENUM.GOLD
+    )
+    return goldToDollarAmount(amount, exchangeRate) || new BigNumber(0)
+  }
+}
+
+function FiatExchangeAmount({ route }: Props) {
+  const { t } = useTranslation(Namespaces.fiatExchangeFlow)
+
+  const [showingMinAmountDialog, setShowingMinAmountDialog] = useState(false)
+  const closeMinAmountDialog = () => setShowingMinAmountDialog(false)
+  const [showingDailyLimitDialog, setShowingDailyLimitDialog] = useState(false)
+
+  const [inputAmount, setInputAmount] = useState('')
+  const parsedInputAmount = parseInputAmount(inputAmount, decimalSeparator)
+  const exchangeRatePair = useSelector(exchangeRatePairSelector)
+  const localCurrencyExchangeRate = useSelector(getLocalCurrencyExchangeRate)
+  const localCurrencyCode = useLocalCurrencyCode()
+  const currencySymbol = LocalCurrencySymbol[localCurrencyCode]
+
+  const currency = route.params.currency
+  const isCusdCashIn = currency === CURRENCY_ENUM.DOLLAR
+  const inputCurrencySymbol = isCusdCashIn ? currencySymbol : ''
+
+  const dollarAmount = useDollarAmount(currency, parsedInputAmount)
+  const localCurrencyAmount = convertDollarsToLocalAmount(dollarAmount, localCurrencyExchangeRate)
+  const dailyLimitCusd = useSelector(cUsdDailyLimitSelector)
+  const minAmountInLocalCurrency = convertDollarsToLocalAmount(
+    DOLLAR_ADD_FUNDS_MIN_AMOUNT,
+    localCurrencyExchangeRate
+  )?.toFixed(0)
+
+  const dispatch = useDispatch()
+
+  React.useEffect(() => {
+    dispatch(fetchExchangeRate())
+  }, [])
+
+  function isNextButtonValid() {
+    return dollarAmount.isGreaterThan(0)
+  }
+
+  function onChangeExchangeAmount(amount: string) {
+    setInputAmount(amount.replace(inputCurrencySymbol, ''))
+  }
+
+  function goToProvidersScreen() {
+    navigate(Screens.ProviderOptionsScreen, {
+      isCashIn: true,
+      currency: route.params.currency,
+      amount: localCurrencyAmount?.toNumber() || 0,
+    })
+  }
+
+  function onPressContinue() {
+    if (dollarAmount.isLessThan(DOLLAR_ADD_FUNDS_MIN_AMOUNT)) {
+      setShowingMinAmountDialog(true)
+      return
+    }
+    if (dollarAmount.isGreaterThan(dailyLimitCusd)) {
+      setShowingDailyLimitDialog(true)
+      return
+    }
+
+    goToProvidersScreen()
+  }
+
+  const closeMinAmountDialogAndContinue = () => {
```

Comment:
I think this should be named `closeDailyLimitDialogAndContinue`

Reply:
🤦‍♂️